packages/twig/src/adapter.js

```diff
@@ -22,6 +22,10 @@ class TwigAdapter extends Fractal.Adapter {
              */
 
             Twig.Templates.registerLoader('fractal', function (location, params) {
+                if (self.views.length === 0) {
```

Comment:
what does this change do? it feels wrong here, e.g. it does not seem like a specific adapter issue. maybe this should be fixed upstream in the Adapter class or somewhere else entirely? I can't suggest something if I don't understand what behavior this changes though :)

Reply:
@mihkeleidast When fractal is started it'll sometimes throw errors for referenced components because it's trying to resolve paths for views it hasn't yet loaded e.g.
```
[0] ✘ Error: Error: Template @component/widget/icon-link/icon-link.twig not found
```

As the tests are passing now I think this is an issue specific to the twig adapter

Reply:
How can this be reproduced? How is it related to the `@@component` resolving syntax? This feels like a separate issue / fix and out of place here.

Reply:
possible alternative solution (might not work as I don't understand the issue completely) that would go in the resolver: on line 58 (e.g. for the regular context resolver) there are if-clauses that check if components exist etc. shouldn't there be some if clauses around the context renderer on line 43?

Reply:
Here's a minimal example demonstrating the issue: https://github.com/justafish/test-fractal
```
yarn install
yarn start
test-fractal on  main [!] is 📦 v1.0.0 via ⬢ v14.11.0 took 22s 
❯ yarn start
yarn run v1.21.1
$ fractal start --sync
✘ Error: Error: Template @component/widget/icon-link/icon-link.twig not found
┌──────────────────────────────────────────┐
│ Fractal web UI server is running!        │
│──────────────────────────────────────────│

```

re alternative solution - I did look into that but tbh I can't recall why it wasn't working out :smile: but I do think it's related to how the twig adapter is loading templates vs handlebars (and hence the tests for that passing fine)

Reply:
Thanks, I'll try to look at it tomorrow.

Reply:
Debugged this a bit, it seems like a deeper issue somewhere in Fractal core.

Not sure how or why, but in your fractal.config.js if you set the engines like this:
```
fractal.docs.set('engine', twigAdapter);
fractal.components.set('engine', twigAdapter);
```
everything works correctly as far as I can understand.

As far as I can understand, setting the engine by method [forces an early load here](https://github.com/frctl/fractal/blob/main/packages/core/src/entities/source.js#L55), and Fractal has not actually parsed the file system at that stage yet (verified this by adding a log [here](https://github.com/frctl/fractal/blob/main/packages/fractal/src/fractal.js#L104)). If you set the config property, Fractal will load the engine a bit later and the views are loaded by then.

I have no idea why it works this way as it's been documented forever to register the engine via the method. I'll discuss this with my peers and maybe we can figure out what is going on here.

Reply:
oh wow, yep that worked :exploding_head: 

I've removed the twig change from this PR in the meantime so this can move forward at least